src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -207,53 +438,95 @@ t8_forest_write_vtk_via_API (t8_forest_t forest, const char *fileprefix,
-        pvtkCell = pyramid;
-        break;
-      default:
-        SC_ABORT_NOT_REACHED ();
+      if (curved_flag == 0) {
+        switch (element_shape) {
+        case T8_ECLASS_VERTEX:
+          pvtkCell = vertex;
+          break;
+        case T8_ECLASS_LINE:
+          pvtkCell = line;
+          break;
+        case T8_ECLASS_QUAD:
+          pvtkCell = quad;
+          break;
+        case T8_ECLASS_TRIANGLE:
+          pvtkCell = tri;
+          break;
+        case T8_ECLASS_HEX:
+          pvtkCell = hexa;
+          break;
+        case T8_ECLASS_TET:
+          pvtkCell = tet;
+          break;
+        case T8_ECLASS_PRISM:
+          pvtkCell = prism;
+          break;
+        case T8_ECLASS_PYRAMID:
+          SC_CHECK_ABORT (element_shape != T8_ECLASS_PYRAMID,
+                          "Pyramids are not supported in vtk output");
+        default:
+          SC_ABORT_NOT_REACHED ();
+        }
+      }
+      else {
+        switch (element_shape) {
+        case T8_ECLASS_VERTEX:
+          pvtkCell = vertex;
+          break;
+        case T8_ECLASS_LINE:
+          pvtkCell = quadraticedge;
+          break;
+        case T8_ECLASS_QUAD:
+          pvtkCell = quadraticquad;
+          break;
+        case T8_ECLASS_TRIANGLE:
+          pvtkCell = quadratictri;
+          break;
+        case T8_ECLASS_HEX:
+          pvtkCell = quadratichexa;
+          break;
+        case T8_ECLASS_TET:
+          pvtkCell = quadratictet;
+          break;
+        case T8_ECLASS_PRISM:
+          pvtkCell = quadraticprism;
+          break;
+        case T8_ECLASS_PYRAMID:
+          SC_CHECK_ABORT (element_shape != T8_ECLASS_PYRAMID,
+                          "Pyramids are not supported in vtk output");
+        default:
+          SC_ABORT_NOT_REACHED ();
+        }
       }
 
       /* For each element we iterate over all points */
       for (ivertex = 0; ivertex < num_corners; ivertex++, point_id++) {
-        /* We take the element coordinates in vtk order */
-        t8_forest_element_coordinate (forest, itree, element,
-                                      t8_eclass_vtk_corner_number
-                                      [element_shape]
-                                      [ivertex], coordinates);
+        /* Compute the vertex coordinates inside [0,1]^dim reference cube. */
+        if (curved_flag) {
+          t8_curved_element_get_reference_node_coords (element, element_shape,
+                                                       scheme, ivertex,
+                                                       vertex_coords);
+        }
+        else {
+          scheme->t8_element_vertex_reference_coords (element,
+                                                      t8_eclass_vtk_corner_number
+                                                      [element_shape]
+                                                      [ivertex],
+                                                      vertex_coords);
+        }
```

Comment:
Couldnt we call `t8_curved_element_get_reference_node_coords` in any case?
It should have the same effect as `t8_element_vertex_reference_coords` for the vertex that occur in non-curved output?